### DISCUSSION TITLE
a question about machine id

### DISCUSSION THREAD

**Discussion by mengyaommm** [hjpotter92/sonyflake-py#7] [Q&A]
![image](https://user-images.githubusercontent.com/40377342/185041194-9603cdc9-753f-4f8c-a4c4-3567af8fd140.png)

hello,  I know the ip_bytes is the binary repressntation of this address，but why the red box marked represent the lower 16 bits of the ip address, the ip is ipv6 , whether it needs to be changed as ip_bytes[14] << 8 + ip_bytes[15]

**Comment by hjpotter92** [comment]
> hello, I know the `ip_bytes` is the binary repressntation of this address，but why the red box marked represent the lower 16 bits of the ip address

I don't understand what you mean here? For an IPv4 of the form `A.B.C.D`, the lowest 16-bits (2-bytes) would be the `C.D` part.

> the ip is ipv6 , whether it needs to be changed as `ip_bytes[14] << 8 + ip_bytes[15]`

The ip should always be an IPv4 value. [`gethostbyname`][1] is documented to only generate the ipv4 value.

[1]: https://docs.python.org/3/library/socket.html#socket.gethostbyname
